### PR TITLE
React Query Set Up

### LINKED ISSUE #46: Data from backend needs to be shown on frontend
**Is your feature request related to a problem? Please describe.**
Data on the frontend currently is hardcoded. We want to allow users to be able to edit, modify, or create a new post. This would require collaboration with the front and backend.

**Describe the solution you'd like**
Set up HTTPS requests so the frontend can call endpoints defined on the backend to get data

### PR DESCRIPTION
**Pull request by StacL**
This closes #46 

## Description / Changes Made
- React Query set up for use (developers can use `axios` **(preferred)** or `fetch` to make a HTTP requests) so that developers can make async calls to get and modify data in our backend (in firebase) to be used for the frontend display
- Created example Query Request to be used for GET requests
- Created example Mutation Request to be used for POST, PUT, and DELETE requests
- Added a date util function to standardize future API calls that need the date

## How to Test
1. Start backend on local 
2. Add `EXPO_PUBLIC_IP_ADDR=<your ip addy>` to your frontend `.env`
3. Add the `SampleMutationRequest` or `SampleQueryRequest` to a page and click the button to send a request

For example, since the friends page is unused currently, I added
```
import React from 'react';
import { SafeAreaView } from 'react-native';
import { Text, View } from 'tamagui';
import SampleQueryRequest from '../../components/SampleQueryRequest'
import SampleMutationRequest from '../../components/SampleMutationRequest'

const Friends = () => {
  return (
    <SafeAreaView>
      <View>
        <Text>Friends/search page</Text>
        <SampleQueryRequest/>
        <SampleMutationRequest/>
      </View>
    </SafeAreaView>
  );
```

4. **NOTE: the results are cached by react-query. If you are testing on android emulator, wipe your data if needed or you can invalidate the cache. If working on browser, open developer tools/inspect and right-click refresh button at top left, then click `empty cache and hard reload`**
5. ** may also need to specify port number to ensure front and backend ports are the same **

## Portions of this code that utilized AI generation
- I used ChatGPT to create a util function that returns the date and time in a specific format (DD-MM-YYYY)
Link: https://chat.openai.com/share/c2c6fe50-55ff-4e8a-885c-4040c6c9dc99

## Sample Pages
Mutation Example
image

Query Example
image

## Checklist
- [x] I have added/updated relevant documentation, and I have followed the coding style guidelines.
- [ ] I have added/updated tests, and I have run the test suite and all tests pass.
- [x] I have checked for any potential conflicts with other branches and fixed any merge conflicts.

### CODE REVIEW

File: frontend/components/SampleMutationRequest.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  const mutation = useMutation({
+    // mutationKey: ['UniqueNameSpecificToMutation'], // Optional: Descriptive key to identify this specific mutation
+    mutationFn: createPost, // Function that defines how to fetch data for this mutation
+  });
+
+  // on submit, send HTTP request
+  const handleSubmit = async () => {
+    setIsLoading(true);
+    setError(null);
```
Comment (noodaj) [inline frontend/components/SampleMutationRequest.tsx:34]: I believe that React Query already has functions for checking fails, errors, and other. Refer to the documentation for more help.

File: frontend/components/SampleMutationRequest.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+export default function SampleMutationRequest() {
+  const [mutationResult, setMutationResult] = useState(null);
+  const [error, setError] = useState(null);
+  const { getToken } = useAuth();
+
+  // Make POST request with axios
+  const createPost = async (postData) => {
+    try {
+      const token = await getToken();
+      console.log(token);
```
Comment (noodaj) [inline frontend/components/SampleMutationRequest.tsx:18]: Remove this

File: frontend/components/SampleMutationRequest.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    // mutationKey: ['UniqueNameSpecificToMutation'], // Optional: Descriptive key to identify this specific mutation
+    mutationFn: createPost, // Function that defines how to fetch data for this mutation
+  });
+
+  // on submit, send HTTP request
+  const handleSubmit = async () => {
+    if (mutation.isPending) return; // Prevent multiple submissions
+    try {
+      const data = await mutation.mutateAsync({
+        author: 'users/user_2cwMgsX7SwXnnnYJ2piefltKxLO',
```
Comment (noodaj) [inline frontend/components/SampleMutationRequest.tsx:50]: Should use the `token` variable from above instead of hardcoding.

**Review by noodaj — CHANGES_REQUESTED** [review]
Left some comments some console logs and hard coding of tokens. Other than that great progress.

File: frontend/components/SampleMutationRequest.tsx (hunk 3 above)
Comment (StacL) [reply frontend/components/SampleMutationRequest.tsx:50]: User token is different than session token for auth, do we have a specific user id that we might need to fetch?

Comment (StacL) [reply frontend/components/SampleMutationRequest.tsx]: NVM, see changes

File: frontend/components/SampleMutationRequest.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  // on submit, send HTTP request
+  const handleSubmit = async () => {
+    if (mutation.isPending) return; // Prevent multiple submissions
+    try {
+      const data = await mutation.mutateAsync({
+        author: `users/${userId}`,
+        comments: [
+          {
+            author: 'users/user_2cwMgsX7SwXnnnYJ2piefltKxLO',
```
Comment (noodaj) [inline frontend/components/SampleMutationRequest.tsx:52]: You forgot it here.

**Review by noodaj — APPROVED** [review]
Great work!

**Review by mitshelle — APPROVED** [review]
Good job with this pull request! This helps us a lot with getting the frontend hooked up to the backend. The changes implemented improved it a lot as well. This will be a great reference for future calls.

**Pull request merged**